src/components/ListView.js

```diff
@@ -1,8 +1,19 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useCollection } from 'react-firebase-hooks/firestore';
 import { db } from '../lib/firebase';
+import { useHistory } from 'react-router-dom';
+
+const ListView = () => {
+  const history = useHistory();
+
+  useEffect(() => {
+    if (!localStorage.getItem('token')) {
+      history.push('/');
+    } else {
+      history.push('/ListView');
+    }
+  });
```

Comment:
I'm assuming you only want this to render upon initial load? If that is the case we could provide an empty array as a second argument to useEffect.

```
  useEffect(() => {
    if (!localStorage.getItem('token')) {
      history.push('/');
    } else {
      history.push('/ListView');
    }
  }, []);
```

Reply:
Agree with this 😸

Reply:
https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

I got a warning when I added the empty array, so based on this doc I added 'history' as a dependency in the array and it seemed to clear it up. Does anyone know if this makes a difference?